### ISSUE TITLE
run-loop's lipo compatibility check is raising an error when app bundle path contains spaces

### ISSUE THREAD

**Issue by shams-ahmed** [calabash/calabash-ios#656]
calabash seems to always output the following message when i run tests. anyone know to to troubleshoot 

```
^Csd-imac-03:EnergyUATTests qaRESET_BETWEEN_SCENARIOS=1 APP_BUNDLE_PATH='../Build/Products/Debug-iphonesimulator/Voltz.app' cucumber
Feature: End to end scenarios, from registration until final checkout

  Scenario: Register a dual fuel user and complete checkout # features/E2E.feature:3
  undefined method `strip' for nil:NilClass (NoMethodError)
  /Users/qa/.rvm/gems/ruby-2.2.0/gems/run_loop-1.2.0.pre5/lib/run_loop/lipo.rb:85:in `block in info'
  /Users/qa/.rvm/gems/ruby-2.2.0/gems/run_loop-1.2.0.pre5/lib/run_loop/lipo.rb:94:in `block in execute_lipo'
  /Users/qa/.rvm/rubies/ruby-2.2.0/lib/ruby/2.2.0/open3.rb:199:in `popen_run'
  /Users/qa/.rvm/rubies/ruby-2.2.0/lib/ruby/2.2.0/open3.rb:93:in `popen3'
  /Users/qa/.rvm/gems/ruby-2.2.0/gems/run_loop-1.2.0.pre5/lib/run_loop/lipo.rb:93:in `execute_lipo'
  /Users/qa/.rvm/gems/ruby-2.2.0/gems/run_loop-1.2.0.pre5/lib/run_loop/lipo.rb:83:in `info'
  /Users/qa/.rvm/gems/ruby-2.2.0/gems/run_loop-1.2.0.pre5/lib/run_loop/lipo.rb:65:in `expect_compatible_arch'
  /Users/qa/.rvm/gems/ruby-2.2.0/gems/run_loop-1.2.0.pre5/lib/run_loop/core.rb:119:in `expect_compatible_simulator_architecture'
  /Users/qa/.rvm/gems/ruby-2.2.0/gems/run_loop-1.2.0.pre5/lib/run_loop/core.rb:214:in `run_with_options'
  /Users/qa/.rvm/gems/ruby-2.2.0/gems/run_loop-1.2.0.pre5/lib/run_loop/core.rb:793:in `run'
  /Users/qa/.rvm/gems/ruby-2.2.0/gems/calabash-cucumber-0.12.0.pre5/lib/calabash-cucumber/launcher.rb:737:in `block in new_run_loop'
  /Users/qa/.rvm/gems/ruby-2.2.0/gems/calabash-cucumber-0.12.0.pre5/lib/calabash-cucumber/launcher.rb:735:in `times'
  /Users/qa/.rvm/gems/ruby-2.2.0/gems/calabash-cucumber-0.12.0.pre5/lib/calabash-cucumber/launcher.rb:735:in `new_run_loop'
  /Users/qa/.rvm/gems/ruby-2.2.0/gems/calabash-cucumber-0.12.0.pre5/lib/calabash-cucumber/launcher.rb:610:in `relaunch'
  /Users/Shared/Jenkins/workspace/Fusion IOS Testing/Energy/EnergyUATTests/features/support/01_launch.rb:28:in `Before'
```
```
sd-imac-03:EnergyUATTests qa$ calabash-ios version
0.12.0.pre5
sd-imac-03:EnergyUATTests qa$ cucumber --version
1.3.18
```

**Comment by jmoody** [comment]
Please run the following command and paste the results here:

```
$ xcrun lipo -info "../Build/Products/Debug-iphonesimulator/Voltz.app/< executable file >
```

The binary name is probably `Voltz`, but it might not be.  You can check the `Voltz.app/Info.plist` for CFExecutableName.

Also, please post your system settings following the guidelines [1].

- [1] https://github.com/calabash/calabash-ios/wiki#reporting-problems

Thanks!

**Comment by shams-ahmed** [comment]
Non-fat file: ../Build/Products/Debug-iphonesimulator/Voltz.app/Voltz is architecture: x86_64

location: 
/Applications/Xcode.app/Contents/Developer/

version:
Xcode 6.1.1
Build version 6A2008a

calabash version: (NOTE i reverted back to the default version)
0.11.4

curl:
{"app_id":"com.simplifydigital.Voltz","outcome":"SUCCESS","version":"0.11.4","app_name":"Voltz","system":"x86_64","simulator_device":"iPhone","simulator":"","app_version":"1.1","iphone_app_emulated_on_ipad":false,"git":{"revision":"bcc992b","remote_origin":"git@github.com:calabash\/calabash-ios-server.git","branch":"master"},"screen_dimensions":{"sample":1,"height":1136,"width":640,"scale":2},"4inch":true,"iOS_version":"8.1"}sd-imac-03:EnergyUATTests qa$

**Comment by shams-ahmed** [comment]
is there a workaround i could this for version 0.11?

**Comment by jmoody** [comment]
Version 0.11 does not include this compatibility check, so you should not see this error.

I am still unsure why you are seeing this exception; the output of `lipo -info` you posted is parseable by the compatibility code.  Maybe lipo is printing to stderr?  I would need your .app bundle to troubleshoot.  Contact me via email if this is possible.

**Comment by shams-ahmed** [comment]
Hey, thanks for pushing the verbose logs out in version 12.0.0... i managed to find out why the error occurred.. It was due to using path which had whitespace in APP_BUNDLE_PATH. see image for error.

![screen shot 2015-01-08 at 10 23 19](https://cloud.githubusercontent.com/assets/2887472/5662753/5177efa8-9735-11e4-86ab-93d61dd4bf67.png)

**Comment by jmoody** [comment]
https://github.com/calabash/run_loop/pull/101

**Comment by jmoody** [comment]
Fixed with run-loop 1.2.0.
